javaee/api/src/main/java/org/jboss/forge/addon/javaee/servlet/SecurityConstraintDescriptor.java

```diff
@@ -0,0 +1,177 @@
+package org.jboss.forge.addon.javaee.servlet;
+
+import org.jboss.forge.addon.javaee.security.TransportGuarantee;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Descriptor of the security constraint element part of the web.xml
+ *
+ * @author <a href="mailto:ivan.st.ivanov@gmail.com">Ivan St. Ivanov</a>
+ */
+public class SecurityConstraintDescriptor
```

Comment:
We will need to rework a little bit the descriptors part anyway. Two reasons besides the one discussed above:
* The web app descriptor does not support removing a single role or constraint (or anything else). It supports removing all of them
* At the moment the SecurityConstraintType returns a list of WebResourceCollections. However, according to the spec, one security constraint element holds just one web resource collection